### ISSUE TITLE
 Stop instances

### ISSUE THREAD

**Issue by magg** [terraform-providers/terraform-provider-oci#697]
How can I specify the state of my instances to be stopped once they are created?
```
terraform -v
Terraform v0.11.11
+ provider.oci v3.1.1
```
Something like this is should be useful for everyone
```
 terraform oci_core_instance.TFInstance -state="STOPPED"
```

**Comment by alexng-canuck** [comment]
Hi @magg ,

AFAIK, `terraform <Resource Name> -state="<State>"` is not a syntactically valid Terraform command. Are you suggesting that the Terraform command should provide a way to directly modify the state of arbitrary resources? If so, this is feedback that would get more traction in the https://github.com/hashicorp/terraform/issues project.

That being said, a possible improvement to the OCI provider would be to allow specifying the state in your TF configuration like this.

```
resource oci_core_instance TFInstance {
  ...
  power_state = "STOPPED"
}
```

and then running `terraform apply`

**Comment by magg** [comment]
Yes that would nice for terraform to allow that syntax to manage the vm's state but i think that won't happen any time soon.

According to these: they think providers should implement such feature

https://github.com/hashicorp/terraform/issues/1579
https://github.com/terraform-providers/terraform-provider-azurerm/issues/555

In any case that improvement you mentioned sounds good

but how would i turn it back on?
```
resource oci_core_instance TFInstance {
  ...
  power_state = "RUNNING"
}``` and then `terraform apply`?

**Comment by kilokahn** [comment]
The feature is now available in the v3.18.0 of the provider.
